vcd/resource_vcd_vapp_access_control_test.go

```diff
@@ -0,0 +1,337 @@
+		"AccessControlIdentifier3": "AC-Vapp3",
+		"VappName0":                "Vapp-AC-0",
+		"VappName1":                "Vapp-AC-1",
+		"VappName2":                "Vapp-AC-2",
+		"VappName3":                "Vapp-AC-3",
+		"UserName1":                "ac-user1",
+		"UserName2":                "ac-user2",
+		"UserName3":                "ac-user3",
+		"RoleName1":                govcd.OrgUserRoleOrganizationAdministrator,
+		"RoleName2":                govcd.OrgUserRoleVappAuthor,
+		"RoleName3":                govcd.OrgUserRoleCatalogAuthor,
+		"AccessLevel1":             types.ControlAccessFullControl,
+		"AccessLevel2":             types.ControlAccessReadWrite,
+		"AccessLevel3":             types.ControlAccessReadOnly,
+		"UserPassword":             "TO_BE_DISCARDED",
+		"FuncName":                 "TestAccVcdVappAccessControl",
+		"Tags":                     "vapp",
+		"SkipNotice":               "",
+	}
+
+	configText := templateFill(testAccVappAccessControl, params)
+	params["AccessLevel1"] = types.ControlAccessReadWrite
+	params["SharedToEveryone"] = "false"
+	params["EveryoneAccessLevel"] = ""
+	params["FuncName"] = "TestAccVcdVappAccessControl-update"
+	params["SkipNotice"] = "# skip-binary-test: only for updates"
+	updateText := templateFill(testAccVappAccessControl, params)
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+	debugPrintf("#[DEBUG] CREATION CONFIGURATION: %s", configText)
+	debugPrintf("#[DEBUG] UPDATE CONFIGURATION: %s", updateText)
+
+	resourceAC0 := "vcd_vapp_access_control.AC-Vapp0"
+	resourceAC1 := "vcd_vapp_access_control.AC-Vapp1"
+	resourceAC2 := "vcd_vapp_access_control.AC-Vapp2"
+	resourceAC3 := "vcd_vapp_access_control.AC-Vapp3"
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckVappAccessControlDestroy(testConfig.VCD.Org, testConfig.VCD.Vdc, []string{"Vapp-AC-0", "Vapp-AC-1", "Vapp-AC-2", "Vapp-AC-3"}),
+		Steps: []resource.TestStep{
+			// Test creation
+			resource.TestStep{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVcdVappAccessControlExists(resourceAC0, testConfig.VCD.Org, testConfig.VCD.Vdc),
+					testAccCheckVcdVappAccessControlExists(resourceAC1, testConfig.VCD.Org, testConfig.VCD.Vdc),
+					testAccCheckVcdVappAccessControlExists(resourceAC2, testConfig.VCD.Org, testConfig.VCD.Vdc),
+					testAccCheckVcdVappAccessControlExists(resourceAC3, testConfig.VCD.Org, testConfig.VCD.Vdc),
+					resource.TestCheckResourceAttr(resourceAC0, "shared_to_everyone", "true"),
+					resource.TestCheckResourceAttr(resourceAC0, "everyone_access_level", "Change"),
+					resource.TestCheckResourceAttr(resourceAC0, "shared.#", "0"),
+
+					resource.TestCheckResourceAttr(resourceAC1, "shared_to_everyone", "false"),
+					resource.TestCheckResourceAttr(resourceAC1, "shared.#", "1"),
+					testAccFindValuesInSet(resourceAC2, "shared", map[string]string{
```

Comment:
The problem is that we cannot check on exact hash, because on every test referred ID will change and therefore hash will always mutate in testing. Resolving this comment.